server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -555,22 +555,36 @@ public String toString() {
         /**
          * Return the current time used for relative calculations. This is
          * {@link System#nanoTime()} truncated to milliseconds.
+         * <p>
+         * If {@link ThreadPool#ESTIMATED_TIME_INTERVAL_SETTING} is set to 0
```

Comment:
Should we make `ThreadPool#ESTIMATED_TIME_INTERVAL_SETTING` a setting that has `0` as an inclusive lower bound? I am fine with that in a follow up.